File: ESC180/labs/lab8/sort.c
#include <stdio.h>

int bubbleSort(int *arr, int size);

int bubbleSort(int *arr, int size){
    int swap = 1;
    int i = 0;
    
    if (arr == NULL){
        return -1;
    }
    while (swap == 1){
        swap = 0;
        for (i=1;i<size;i++){
            if (arr[i-1] > arr[i]){
                int temp = arr[i-1];
                arr[i-1] = arr[i];
                arr[i] = temp;
                swap = 1;
            }
        }
    }
    return 0;
}

File: ESC180/labs/lab8/sort.py
def bubbleSort(array):
    n = len(array)
    if (n == 0):
        return [-1,array]
    swap = True
    
    while (swap):
        swap = False
        for i in range(1,n):
            if (array[i-1]>array[i]):
                temp = array[i-1]
                array[i-1] = array[i]
                array[i] = temp
                swap = True
    return [0, array]

    
